gpt_index/readers/chroma.py

```diff
@@ -31,41 +33,88 @@ def __init__(
         except ImportError:
             raise ValueError(import_err_msg)
 
-        if (collection_name is None) or (persist_directory is None):
-            raise ValueError("Please provide a collection name and persist directory.")
-
+        if collection_name is None:
+            raise ValueError("Please provide a collection name.")
         from chromadb.config import Settings
 
-        self._client = chromadb.Client(
-            Settings(
-                chroma_db_impl="duckdb+parquet", persist_directory=persist_directory
+        if persist_directory:
+            self._client = chromadb.Client(
+                Settings(
+                    chroma_db_impl="duckdb+parquet", persist_directory=persist_directory
+                )
+            )
+        else:
+            self._client = chromadb.Client(
+                Settings(
+                    chroma_api_impl="rest",
+                    chroma_server_host=host,
+                    chroma_server_http_port=port,
+                )
             )
-        )
         self._collection = self._client.get_collection(collection_name)
 
-    def load_data(
-        self,
-        query_vector: Any,
-        limit: int = 10,
-    ) -> Any:
-        """Load data from Chroma.
+    def create_documents(self, results: tuple) -> List[Document]:
+        """Create documents from the results.
 
         Args:
-            query_vector (Any): Query
-            limit (int): Number of results to return.
+            results: Results from the query.
 
         Returns:
-            List[Document]: A list of documents.
+            List of documents.
         """
-        results = self._collection.query(query_embeddings=query_vector, n_results=limit)
-
         documents = []
-        for result in zip(results["ids"], results["documents"], results["embeddings"]):
+        for result in zip(
+            results["ids"],
+            results["documents"],
+            results["embeddings"],
+            results["metadatas"],
+        ):
             document = Document(
                 doc_id=result[0][0],
                 text=result[1][0],
                 embedding=result[2][0],
+                extra_info=result[3][0],
             )
             documents.append(document)
 
         return documents
+
+    def load_data(
+        self,
+        query_embedding: List[float] = None,
+        limit: int = 10,
+        where: dict = {},
```

Comment:
nit: initialize empty dict as a default function argument can be a bit unsafe, i'd do something like `where: Optional[dict] = None`, and then in the function below do `where = where or {}`